TypeSystem/Schema/golden/smoke_test_struct.cc

```diff
@@ -34,20 +34,23 @@ struct Empty {
 struct X {
   int32_t x;
 };
+enum class E : uint16_t {};
 struct Y {
-  int32_t x;
+  E e;
 };
+using MyFreakingVariant = Variant<A, X, Y>;
 struct C {
   Empty e;
-  Variant<X, Y> c;
+  MyFreakingVariant c;
 };
+using Variant_B_A_B_C_Empty_E = Variant<A, B, C, Empty>;
```

Comment:
It's analogous to C / C++ name mangling by the linker. Purpose: for name of `Variant<Variant<Foo, Bar>, Baz>` to make sense.